### PR TITLE
fix: updates to required JS projects

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Related to https://github.com/freeCodeCamp/freeCodeCamp/pull/52370

### CODE REVIEW

**Comment by jdwilkin4** [comment]
Hey @scissorsneedfoodtoo ! 

I will have more time to review this later on this week.
But one thing I have noticed in these projects is that we are using `innerHTML` for everything when we really should reserve that for when we are setting new HTML. If we are just setting text, then we should use another method like `textContent`.

**Comment by scissorsneedfoodtoo** [comment]
Hey @jdwilkin4, thanks for taking a quick look at this and catching the issue with using `innerHTML` rather than `innerText`. That's something I missed while originally reviewing these projects.

I can go in and fix those, and apply the updated code to the Learn PR solutions.

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Looks good.
Left a couple of comments 👍

File: apps/pokemon-search-app/index.html [hunk 1]
```diff
@@ -38,7 +30,7 @@ <h1>Pokémon Search App</h1>
             <span id="weight"></span>
             <span id="height"></span>
           </div>
-          <img src alt id="sprite" />
+          <div class="sprite-container"><img alt id="sprite" /></div>
```
[the remark this answers is not in the document]
Comment (scissorsneedfoodtoo) [reply apps/pokemon-search-app/index.html:33]: Thanks for taking a look at this, @jdwilkin4, and for running the code though a validator. That's a great idea – I've just been using Prettier this whole time, not checking for actual validation errors.

We did have a default image before with a Pikachu sprite, but it was decided that we should remove that just to be extra safe.

We could go with a black box there, just a `div` with a background color and swap it out for an `img` element whenever appropriate. Or we could just add an `img` there whenever we get a response from the API proxy and remove it whenever there are errors.

I'll play around with it and push an update ASAP.

[the remark this answers is not in the document]
Comment (scissorsneedfoodtoo) [reply apps/pokemon-search-app/index.html]: This should be fixed in the demo here, and in the solution in the Learn PR.

File: apps/pokemon-search-app/public/script.js [hunk 2]
```diff
@@ -22,57 +22,52 @@ const getPokemon = async () => {
...
-});
-
-function resetDisplay() {
-  // reset to default display if pokemon is not found
-
+const resetDisplay = () => {
   // reset image src and alt
-  sprite.src = "";
-  sprite.alt = "";
+  sprite.src = '';
```
[the remark this answers is not in the document]
Comment (scissorsneedfoodtoo) [reply apps/pokemon-search-app/public/script.js]: This should be fixed in the demo here, and in the solution in the Learn PR.

**Review by jdwilkin4 — APPROVED** [review]
Looks good 👍

**Review by naomi-lgbt — COMMENTED** [review]
Couple of nits

File: apps/palindrome-checker/public/script.js [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
 
   if (input === '') {
     alert('Please input a value');
     return;
   }
 
-  if (hasSpecialCharactersOrDigits === true) {
+  if (hasSpecialCharactersOrDigits) {
     alert('Input should not include numbers and special characters');
     return;
```
[the remark this answers is not in the document]
Comment (jdwilkin4) [reply apps/palindrome-checker/public/script.js:16]: I think @scissorsneedfoodtoo mentioned there was some sort of issue about having them strip the characters. 
There was a prior discussion thread in the old staff chat but I don't remember the exact reason why this was a problem.
So @scissorsneedfoodtoo should have the details on that.

[the remark this answers is not in the document]
Comment (scissorsneedfoodtoo) [reply apps/palindrome-checker/public/script.js:16]: The last messages in chat were about the demo project incorrectly flagging spaces as special characters. I can't remember exactly why we have this alert for special characters, or when it got added.

I'm in favor of updating the project so it's closer to the current palindrome checker, where special characters and spaces are stripped out before checking for a palindrome. It shouldn't be too difficult to update the solution here and add a couple of extra tests.

File: apps/roman-numeral-converter/index.html
[the remark this answers is not in the document]
Comment (scissorsneedfoodtoo) [reply apps/roman-numeral-converter/index.html:1]: Done! It didn't take all that long to swap some of the colors around. I'll post screenshots of the latest Roman Numeral Converter styling below so they're easier for everyone to see.

**Comment by scissorsneedfoodtoo** [comment]
Here's what the dark mode version of the Roman Numeral Converter project looks like:

image

image

**Comment by scissorsneedfoodtoo** [comment]
Added a logo and adjusted the styling of the Pokémon Search app slightly:

image

File: apps/palindrome-checker/public/script.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (scissorsneedfoodtoo) [reply apps/palindrome-checker/public/script.js]: The Palindrome Check here and in https://github.com/freeCodeCamp/freeCodeCamp/pull/52370 have been updated so they're closer to the solution we have in the current JS certification where whitespace and special characters are stripped before checking for a palindrome.

**Review by naomi-lgbt — APPROVED** [review]
image

Okay, everything is looking good on my end. Let's go ahead and land this. Thanks for all your work on these!

**Pull request merged**